bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java

```diff
@@ -94,76 +94,103 @@
 
     // Pairing interval in seconds
     private static final int PAIRING_INTERVAL = 10;
+    private static final int CONNECT_TIMEOUT = 10;
 
     private final Logger logger = LoggerFactory.getLogger(NanoleafControllerHandler.class);
+    private HttpClientFactory httpClientFactory;
     private HttpClient httpClient;
-    private List<NanoleafControllerListener> controllerListeners = new CopyOnWriteArrayList<>();
 
-    // Pairing, update and panel discovery jobs and touch event job
+    private @Nullable HttpClient httpClientSSETouchEvent;
+    private @Nullable Request sseTouchjobRequest;
+    private List<NanoleafControllerListener> controllerListeners = new CopyOnWriteArrayList<NanoleafControllerListener>();
+
     private @NonNullByDefault({}) ScheduledFuture<?> pairingJob;
     private @NonNullByDefault({}) ScheduledFuture<?> updateJob;
     private @NonNullByDefault({}) ScheduledFuture<?> touchJob;
-
-    // JSON parser for API responses
     private final Gson gson = new Gson();
 
-    // Controller configuration settings and channel values
     private @Nullable String address;
     private int port;
     private int refreshIntervall;
     private @Nullable String authToken;
     private @Nullable String deviceType;
     private @NonNullByDefault({}) ControllerInfo controllerInfo;
 
-    public NanoleafControllerHandler(Bridge bridge, HttpClient httpClient) {
+    private boolean touchJobRunning = false;
+
+    public NanoleafControllerHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
         super(bridge);
-        this.httpClient = httpClient;
+        this.httpClientFactory = httpClientFactory;
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+        this.httpClient.setConnectTimeout(CONNECT_TIMEOUT * 1000L);
```

Comment:
That looks like a major flaw then. I better go through all the code and fix that and make sure this is applied to the request directly.

For some reason I cannot comment on the above comment: "So why didn't you apply it then" so I answer her

![image](https://user-images.githubusercontent.com/5937600/134294882-d4179cfc-475f-40b6-a31e-986c0811ee2a.png)

Actually I  don't know what you are recommending here. I added the notation to suppress the Warning but I can do better now as from what I learned above from "pacive" lately (though I hate this boilerplate to circumvent the warnings)

![image](https://user-images.githubusercontent.com/5937600/134295590-51685ea9-2da5-4860-b231-8facc930ac0c.png)

Reply:
`Afaik, you should not set this here on the common client as it will effect everyone using this client.
Instead, you can set timeouts on your specific requests that you are creating.`

I checked the code. There general connect-timeout on the main httpClient that you referred to was actually superfluous as the timeout is already set in the central OpenAPI.requestBuilder() - method for each request, so I just followed your recommendation to drop that here.

Reply:
> Actually I don't know what you are recommending here. I added the notation to suppress the Warning but I can do better now as from what I learned above from "pacive" lately

Yes, applying the way from pacive to avoid the suppression, that was my recommendation there.